include/rmm/device_scalar.hpp

```diff
@@ -73,15 +73,32 @@ class device_scalar {
     return host_value;
   }
 
+  /**---------------------------------------------------------------------------*
+   * @brief Copies the value from hostto device.
+   *
+   * @param host_value The value of the scalar after synchronizing its stream
+   *---------------------------------------------------------------------------**/
+  void set_value(T host_value) {
+    auto status = cudaMemcpyAsync(buff.data(), &host_value, sizeof(T),
+                                  cudaMemcpyDefault, buff.stream());
```

Comment:
What if I want to do the copy on a different stream?

Reply:
Makes me wonder, should we ever default to the buffer's stream, or just use the default stream = 0?

Reply:
I ask because the assignment operators on device_buffer can't allow the user to specify the stream. Is that the exception, and when do we make that exception?